ext/standard/uniqid.c

```diff
@@ -53,6 +87,24 @@ PHP_FUNCTION(uniqid)
 		Z_PARAM_BOOL(more_entropy)
 	ZEND_PARSE_PARAMETERS_END();
 
+#ifdef HAVE_UUIDGEN
+	struct uuid uuid;
+	int *n = (int *)&uuid;
+
+	/* Use faster uuidgen() if available */
+	(void)uuidgen(&uuid, 1);
+
+	if (more_entropy) {
+		n[1] &= 0xffffff;
+		uniqid = strpprintf(0, "%s%08x%06x.%08x", prefix, n[0], n[1], n[2]);
+	} else {
```

Comment:
This output format is incompatible with the non-uuidgen output. It also leaks parts of the hosts MAC address and also doesn't actually provide “more entropy”, since the mac address is fixed on a single node.

Reply:
It is actually compatible, despite the fact I do not understand how gettimeofday+more entropy produces 14+8 characters, where I would expect 13+8 reading the code. Check it on your own:
$ php -r 'printf("%s\n", uniqid("", 1));'
67ee7f958a5b34.94453710

For the MAC address, I understand it is not included in UUID anymore:
$ uuidgen       
bf0ff2ae-cf7a-46b1-bd2e-c3577268e311
$ uuidgen 
39638058-e887-41a5-97d1-6451b0622763

But I can understand it may be platform dependent, and we must avoid using it in case it could contains the MAC.